Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -368,7 +364,37 @@ extension AWSClient {
             middlewares: middlewares
         )
     }
+
+    fileprivate func urlEncodeQueryParams(fromDictionary dict: [String:Any]) -> String? {
+        var components = URLComponents()
+        components.queryItems = urlQueryItems(fromDictionary: dict)
+        if components.queryItems != nil, let url = components.url {
+            return url.query
+        }
+        return nil
+    }
+
+    fileprivate func urlQueryItems(fromDictionary dict: [String:Any]) -> [URLQueryItem]? {
+        var queryItems: [URLQueryItem] = []
+        let keys = Array(dict.keys).sorted()
+
+        for key in keys {
+            if let value = dict[key] {
+                queryItems.append(URLQueryItem(name: key, value: String(describing: value)))
+            }
+        }
+        return queryItems.isEmpty ? nil : queryItems
+    }
+}
+
+// debug request creator
+#if DEBUG
```

Comment:
whoa, nice

Reply:
thnx.  was actually worried this might be poorly received :)